lib/extensions.js

```diff
@@ -477,4 +483,48 @@ module.exports = function(realmConstructor, context) {
         },
         configurable: true
     });
-}
+
+    Object.defineProperty(realmConstructor, "JsonSerializationReplacer", {
+        get: function () {
+            const seen = [];
+
+            return function(_, value) {
+                // Only check for circular references when dealing with objects & arrays.
+                if (value === null || typeof value !== "object") {
+                    return value;
+                }
+
+                // 'this' refers to the object or array containing the the current key/value.
+                const parent = this;
+
+                // Expose the non-enumerable prop $refId for circular serialization, if it exists.
+                if (value.$refId) {
+                    Object.defineProperty(value, "$refId", { enumerable: true })
+                }
+
+                if (!seen.length) {
+                    // If we haven't seen anything yet, we push the current value (root element/array).
+                    seen.push(value);
```

Comment:
you could do `return value;` here and avoid the big nested `else` after it. The `return early` style ;). Of course that's not always possible but here it seems it will be alright.